batchflow/config.py

```diff
@@ -1,289 +1,245 @@
         """
-        if isinstance(config, Config):
-            val = config.get(variables, default=default)
-        else:
-            val = self._get(variables, config, default=default, pop=False)
-        return val
+        value = self._get(key, default=default)
+        return value
+    
+    def pop(self, key, **kwargs):
+        """ Returns the value or tuple of values for key and remove them from config
+
+        Parameters
+        ----------
+        key : str or list of hashable objects
+            '/' is used to get value from nested dict
+
+        Returns
+        -------
+        single value or a tuple
+        """
+        value = self._get(key, pop=True, **kwargs)
+        return value
+    
+    def _get(self, key, **kwargs):
 
-    def _get(self, variables, config=None, **kwargs):
-        if config is None:
-            config = self.config
         pop = kwargs.get('pop', False)
-        has_default = 'default' in kwargs
+        has_default = 'default' in kwargs 
         default = kwargs.get('default')
+        default = Config(default) if isinstance(default, dict) else default
+
+        method = super().get if not pop else super().pop
 
         unpack = False
-        if not isinstance(variables, (list, tuple)):
-            variables = list([variables])
+        if not isinstance(key, list):
+            key = list([key])
             unpack = True
 
         ret_vars = []
-        for variable in variables:
-            _config = config
-            if '/' in variable:
-                var = variable.split('/')
-                prefix = var[:-1]
-                var_name = var[-1]
-            else:
-                prefix = []
-                var_name = variable
-
-            for p in prefix:
-                if p in _config:
-                    _config = _config[p]
-                else:
-                    _config = None
-                    break
-            if isinstance(_config, dict):
-                if pop:
-                    if has_default:
-                        val = _config.pop(var_name, default)
+        for variable in key:
+            if isinstance(variable, str) and '/' in variable:
+                keys = variable.split('/')
+                value = self
+                for k in keys:
+                    if isinstance(value, dict):
+                        parent = value
+                        value = value[k]
                     else:
-                        val = _config.pop(var_name)
+                        if has_default:
+                            return default
+                        raise KeyError(k)
+                if pop:
+                    del parent[k]
+
+            else:
+                if variable in self:
+                    value = method(variable)
                 else:
                     if has_default:
-                        val = _config.get(var_name, default)
+                        return default
                     else:
```

Comment:
extra `else` here.
Please, provide some comments, because this method is hard to read and understand.